From 376d899148c5dc20a164d2c51bd9a277618022b9 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Wed, 3 Apr 2013 14:47:10 +0200
Subject: [PATCH 0996/1083] drm/rcar-du: Use the mode active size in mode set

The driver currently configures the dislay to the frame buffer size
while it should use the requested mode active size. Fix it.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 286d0dc3fb6a340b9630ddc5463bcfaa983cbb3a)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 7eafbccf2e15fc6a52d6f845a1eeddbf2ebead0d)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index db53c25..53dbe36 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -412,8 +412,8 @@ static int rcar_du_crtc_mode_set(struct drm_crtc *crtc,
 
 	rcrtc->plane->src_x = x;
 	rcrtc->plane->src_y = y;
-	rcrtc->plane->width = crtc->fb->width;
-	rcrtc->plane->height = crtc->fb->height;
+	rcrtc->plane->width = mode->hdisplay;
+	rcrtc->plane->height = mode->vdisplay;
 
 	rcar_du_plane_compute_base(rcrtc->plane, crtc->fb);
 
-- 
1.8.3.2

